order.py

```diff
@@ -0,0 +1,216 @@
+                self.orderSells[self.SellPrices.index(order.price)].append(order.orderId)
+            else:
+                price_ind = bisect.bisect(self.SellPrices, order.price)
+                self.SellPrices.insert(price_ind, order.price)
+                self.orderSells.insert(price_ind, [order.orderId])
+        
+        return(self.orderProcess(order))
+          
+    def makeTrade(self, buy_order, sell_order):
+        fill_qty = min(buy_order.qty, sell_order.qty)
+        buy_order.qty -= fill_qty
+        sell_order.qty -= fill_qty
+        fill_id = self.new_fillId()
+        fill_timestamp = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time.time()))
+        fill_msg_buyer = fillMessage(buy_order.orderId, fill_id, fill_timestamp, fill_qty)
+        fill_msg_seller = fillMessage(sell_order.orderId, fill_id, fill_timestamp, fill_qty)
+        ## Still need to send fill_msg out to traders !!!
+        
+        if buy_order.qty == 0:
+            loc_buyPrice = self.BuyPrices.index(buy_order.price)
+            loc_order = self.orderBuys[loc_buyPrice].index(buy_order.orderId)
+            self.orderBuys[loc_buyPrice].pop(loc_order)
+            if len(self.orderBuys[loc_buyPrice]) == 1:
+                self.BuyPrices.pop(loc_buyPrice)
+            self.orderMap.pop(buy_order.orderId)
+            self.standingBuys.pop(self.standingBuys.index(buy_order.orderId))
+            
+        if sell_order.qty == 0:
+            loc_sellPrice = self.SellPrices.index(sell_order.price)
+            loc_order = self.orderBuys[loc_sellPrice].index(sell_order.orderId)
+            self.orderMap.pop(sell_order.orderId)
+            if len(self.orderSells[loc_sellPrice]) == 1:
+                self.SellPrices.pop(loc_sellPrice)
+            self.orderMap.pop(sell_order.orderId)
+        
+        return([buy_order, sell_order])
+                           
+            
+    def orderProcess(self, recent_order):
+        if recent_order.side == "buy" and len(self.SellPrices)>0 and self.SellPrices[0] <= recent_order.price:
+            while recent_order.qty > 0:
+                cheapest_sell = self.orderMap[self.orderSells[0][0]]
+                recent_order = self.makeTrade(recent_order, cheapest_sell)[0]
+                
+        elif recent_order.side == "sell" and len(self.BuyPrices)>0 and self.BuyPrices[-1] >= recent_order.price:
+            while recent_order.qty > 0:
+                first_buy_ind = self.orderSells[bisect.bisect_left(self.BuyPrices, recent_order.price)][0]
+                recent_order = self.makeTrade(self.orderMap[first_buy_ind], recent_order)[1]
+                
+        else: # if no trade can be made, send Acknowledgement, otherwise fillMessage
+            order_timestamp = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time.time()))
+            new_orderAck = orderAcknowledgement(recent_order.orderId, order_timestamp)
+            self.msgs.append(new_orderAck)
+            return(new_orderAck)

```

Comment:
We always want to ack an order right? This looks like if a sell or buy can be made we don't actually ack the order.